### PR TITLE
feat(datetime): datetime manual testing

### PR DESCRIPTION
**Pull request by timrombergjakobsson**

### CODE REVIEW

File: src/components/Datetime.tsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
+import { useState, useRef } from 'react';
+
+const Datetime = () => {
+  const [typeOfState, setTypeOfState] = useState<'success' | 'error'>('success');
+  const [typeOfSize, setTypeOfSize] = useState<'sm' | 'md' | 'lg'>('lg');
+  const [typeOfInput, setTypeOfInput] = useState<'datetime-local' | 'date' | 'time'>('datetime-local');
+  const tdsDatetimeRef = useRef<HTMLTdsDatetimeElement>(null);
```
Comment (alexandereneroth) [inline src/components/Datetime.tsx:7]: This appears unused

File: src/components/Datetime.tsx [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+  const [noMinWidth, setNoMinWidth] = useState(false); // State for noMinWidth
+	const [isDatetimeDisabled, setIsDatetimeDisabled] = useState(false);
+
+  const handleNewState = (event: Event) => {
+    const newState: 'error' | 'success' = (event.target as HTMLTdsRadioButtonElement).value as 'error' | 'success';
+    setTypeOfState(newState);
+  };
```
Comment (alexandereneroth) [inline src/components/Datetime.tsx:14]:
```suggestion
  const handleNewState = (event: Event) => {
    const newState = (event.target as HTMLTdsRadioButtonElement).value as 'error' | 'success';
    setTypeOfState(newState);
  };
```

The type on newState is not necessary as you're already casting the value to the same type. (same in other event handlers)

File: src/components/Datetime.tsx [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+  const handleSize = (event: Event) => {
+    const newSize: 'sm' | 'md' | 'lg' = (event.target as HTMLTdsRadioButtonElement).value as 'sm' | 'md' | 'lg';
+    setTypeOfSize(newSize);
+  };
+
+	const handleToggle = (event: CustomEvent) => {
+		const { detail } = event as CustomEvent<{ checked: boolean }>;
+		setNoMinWidth(detail.checked);
+	};
```
Comment (alexandereneroth) [inline src/components/Datetime.tsx:29]:
```suggestion
	const handleToggle = (event: CustomEvent<{ checked: boolean }>) => {
		const { detail } = event;
		setNoMinWidth(detail.checked);
	};
```

Simpler ☝️

File: src/components/Datetime.tsx [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+
+	const handleDisabledState = (event: CustomEvent) => {
+		const { checked } = event.detail;
+		setIsDatetimeDisabled(checked);
+	};
```
Comment (alexandereneroth) [inline src/components/Datetime.tsx:34]:
```suggestion
	const handleDisabledState = (event: CustomEvent<{ checked: boolean }>) => {
		const { detail } = event;
		setIsDatetimeDisabled(detail.checked);
	};
```

To make it consistent with the event handler directly above. 👆

File: src/components/Datetime.tsx [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+      <div className="tds-headline-02 tds-u-pb1">Datetime</div>
+      <div>
+        <div className="tds-headline-04 tds-u-pb1">State</div>
+        <tds-radio-button
+          name="success"
+          value="success"
+          checked={typeOfState === 'success'}
+          ref={element => {
+            if (element) {
+              element.addEventListener('tdsChange', handleNewState as unknown as EventListener);
```
Comment (alexandereneroth) [inline src/components/Datetime.tsx:47]:
```suggestion
              element.addEventListener('tdsChange', handleNewState);
```

Doesn't look like you need to cast theese.

**Review by alexandereneroth — CHANGES_REQUESTED** [review]
Nice work!
Left some feedback 🙂

**Review by mJarsater — COMMENTED** [review]
Nice, maybe add a datetime that uses the default-value attribute? Maybe in the form page, set to todays date using the default-value.

File: src/components/Datetime.tsx (hunk 4 above)
Comment (timrombergjakobsson) [reply src/components/Datetime.tsx:34]: Nice, I agree with the feedback!

**Comment by mJarsater** [comment]
> Nice, maybe add a datetime that uses the default-value attribute? Maybe in the form page, set to todays date using the default-value.

Just realised we already had the datetime in there.. my bad. Maybe you could update it to always display todays date?

**Comment by timrombergjakobsson** [comment]
> > Nice, maybe add a datetime that uses the default-value attribute? Maybe in the form page, set to todays date using the default-value.
> 
> Just realised we already had the datetime in there.. my bad. Maybe you could update it to always display todays date?

You mean use the `value={}` on the component? It is that one that you can use when you wanna set it?

**Comment by mJarsater** [comment]
> showing the cur

Alright, I think we can use the default-value prop. I can have a look locally

**Comment by timrombergjakobsson** [comment]
> > > > Nice, maybe add a datetime that uses the default-value attribute? Maybe in the form page, set to todays date using the default-value.
> > > 
> > > 
> > > Just realised we already had the datetime in there.. my bad. Maybe you could update it to always display todays date?
> > 
> > 
> > You mean use the `value={}` on the component? It is that one that you can use when you wanna set it?
> 
> > Hm trying it out like this:
> 
> ```
> const [currentDate, setCurrentDate] = useState('');
> 
>   useEffect(() => {
>     // Get current date and format it
>     const today = new Date();
>     const year = today.getFullYear();
>     const month = String(today.getMonth() + 1).padStart(2, '0');
>     const day = String(today.getDate()).padStart(2, '0');
>     const formattedDate = `${year}-${month}-${day}`;
>   
>     setCurrentDate(formattedDate);
>   }, []);
> ```
> 
> > And then:
> > ` value={currentDate}`
> > This however is not showing the current date? Maybe doing something weird here.
> 
> Tried it out locally and it worked for me when I did:
> 
> ```
> <tds-datetime
>     mode-variant="secondary"
>     label="Date of birth"
>     name="dateOfBirth"
>     type="date"
>     default-value={currentDate}
>   ></tds-datetime>
> ```

It does? And you using the same useEffect code I pasted?

**Comment by timrombergjakobsson** [comment]
> <tds-datetime
>     mode-variant="secondary"
>     label="Date of birth"
>     name="dateOfBirth"
>     type="date"
>     default-value={currentDate}
>   ></tds-datetime>

Ahh ok I cant use the dynamic type for example.

**Comment by mJarsater** [comment]
> > > > > Nice, maybe add a datetime that uses the default-value attribute? Maybe in the form page, set to todays date using the default-value.
> > > > 
> > > > 
> > > > Just realised we already had the datetime in there.. my bad. Maybe you could update it to always display todays date?
> > > 
> > > 
> > > You mean use the `value={}` on the component? It is that one that you can use when you wanna set it?
> > 
> > 
> > > Hm trying it out like this:
> > 
> > 
> > ```
> > const [currentDate, setCurrentDate] = useState('');
> > 
> >   useEffect(() => {
> >     // Get current date and format it
> >     const today = new Date();
> >     const year = today.getFullYear();
> >     const month = String(today.getMonth() + 1).padStart(2, '0');
> >     const day = String(today.getDate()).padStart(2, '0');
> >     const formattedDate = `${year}-${month}-${day}`;
> >   
> >     setCurrentDate(formattedDate);
> >   }, []);
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > > And then:
> > > ` value={currentDate}`
> > > This however is not showing the current date? Maybe doing something weird here.
> > 
> > 
> > Tried it out locally and it worked for me when I did:
> > ```
> > <tds-datetime
> >     mode-variant="secondary"
> >     label="Date of birth"
> >     name="dateOfBirth"
> >     type="date"
> >     default-value={currentDate}
> >   ></tds-datetime>
> > ```
> 
> It does? And you using the same useEffect code I pasted?

Sorry, no. I moved it out of the useEffect. Since it's just a default value it won't have to use any state. So I just did:

```
const FormPage = () => {
  const today = new Date();
  const year = today.getFullYear();
  const month = String(today.getMonth() + 1).padStart(2, '0');
  const day = String(today.getDate()).padStart(2, '0');
  const formattedDate = `${year}-${month}-${day}`;

return (
<tds-datetime
                mode-variant="secondary"
                label="Date of birth"
                name="dateOfBirth"
                type="date"
                default-value={formattedDate}
              ></tds-datetime>
)
```

File: src/components/Datetime.tsx [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+    setTypeOfState(newState);
+  };
+
+  const handleType = (event: Event) => {
+    const newTypeOfInput: 'datetime-local' | 'date' | 'time' = (event.target as HTMLTdsRadioButtonElement).value as 'datetime-local' | 'date' | 'time';
+    setTypeOfInput(newTypeOfInput);
+  };
+
+  const handleSize = (event: Event) => {
+    const newSize: 'sm' | 'md' | 'lg' = (event.target as HTMLTdsRadioButtonElement).value as 'sm' | 'md' | 'lg';
```
Comment (alexandereneroth) [inline src/components/Datetime.tsx:29]:
```suggestion
    const newSize = (event.target as HTMLTdsRadioButtonElement).value as 'sm' | 'md' | 'lg';
```

Same here as you did above.

File: src/components/Datetime.tsx [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+  const formattedDate = `${year}-${month}-${day}`;
+  
+
+  const handleNewState = (event: Event) => {
+    const newState = (event.target as HTMLTdsRadioButtonElement).value as 'error' | 'success';
+    setTypeOfState(newState);
+  };
+
+  const handleType = (event: Event) => {
+    const newTypeOfInput: 'datetime-local' | 'date' | 'time' = (event.target as HTMLTdsRadioButtonElement).value as 'datetime-local' | 'date' | 'time';
```
Comment (alexandereneroth) [inline src/components/Datetime.tsx:24]:
```suggestion
    const newTypeOfInput = (event.target as HTMLTdsRadioButtonElement).value as 'datetime-local' | 'date' | 'time';
```

also here

**Review by alexandereneroth — APPROVED** [review]
nice

**Review by alexandereneroth — CHANGES_REQUESTED** [review]
See the last few changed i posted previously!

**Review by alexandereneroth — APPROVED** [review]
🎇

**Review by mJarsater — COMMENTED** [review]
🧨

**Review by mJarsater — APPROVED** [review]
🧨

**Pull request merged**